api/indexer/tx_indexer.go

```diff
@@ -27,8 +29,12 @@ var (
 	_ event.Subscription[*chain.StatelessBlock]        = (*txDBIndexer)(nil)
 )
 
-func WithIndexer(name string, path string) vm.Option {
-	return func(v *vm.VM) error {
+func With(name string, path string) vm.Option {
+	return vm.NewOption(Namespace, OptionFunc(name, path))
+}
+
+func OptionFunc(name string, path string) vm.OptionFunc {
```

Comment:
future PR: I think we can remove name and path from the tx indexer and make them constants in this package